strawberry/object_type.py

```diff
@@ -65,10 +77,10 @@ def _check_field_annotations(cls: Type[Any]):
         # to make sure dataclasses.dataclass is ready for it
         if isinstance(field_, StrawberryField):
             # If the field has a type override then use that instead of using
-            # the class annotations or resolver annotation
+            # the class annotations or resolver annotation, but only if we
+            # don't have an annotation already in there
             if field_.type_annotation is not None:
-                cls_annotations[field_name] = field_.type_annotation.annotation
-                continue
+                replace_annotation(field_name, field_.type_annotation.annotation)
```

Comment:
we do we need to replace the annotation? couldn't we use the original one?

Reply:
I tried but it basically broke the `graphql_type` override functionality. That is probably because the dataclass will set the field's `type` based on the annotation, thus resetting the custom value we passed.

One possibility here is to set a flag to "not override the field's type" during dataclass creation, but it seems too "hacky" to me.

Not sure if there's a better solution for this